heksher/setting.py

```diff
@@ -81,58 +91,39 @@ def get(self, **contexts) -> T:
             return self.default_value
         return from_rules
 
-    def update(self, client, context_features: Sequence[str], root: RuleBranch[T]):
+    def update(self, client, context_features: Sequence[str], rules: Iterable[Tuple[Sequence[Tuple[str, str]], T]],
+               is_rule_collection: bool = True):
         """
         Update a setting's rules from a client
 
         Args:
             client: The client updating the setting.
             context_features: The context features the root was collated by.
-            root: A collated rulebranch root.
+            rules: An iterable of rules to collate.
+            is_rule_collection: bool, if rules are a collection; if not, don't collate the rules
         """
         if self.last_ruleset:
             last_client = self.last_ruleset.client()
             if last_client and last_client is not client:
                 logger.warning('setting received rule set from multiple clients',
                                extra={'setting': self.name, 'new_client': client, 'last_client': last_client})
+        if is_rule_collection:
+            updated_rules: List[Tuple[Sequence[Tuple[str, str]], T]] = []
+            for rule in rules:
+                rule_ = rule[0]
+                value_ = rule[1]
+                for callback in self._callbacks:
+                    value_ = callback(self, rule_, value_)  # type: ignore
+                updated_rules.append((rule_, value_))
+            root = collate_rules(context_features, updated_rules)
+        else:
+            root_ = rules
+            for callback in self._callbacks:
+                root_ = callback(self, '', root_)  # type: ignore
```

Comment:
???

Reply:
how do you run the callbacks when the rules are not a collection? for example, only set as a value 10..?